### PR TITLE
Add allowance

### PR DESCRIPTION
**Pull request by Pattio**

### CODE REVIEW

File: Sources/Entities/Allowance/PSAllowanceValid.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import ObjectMapper
+
+public class PSAllowanceValid: Mappable {
+    public var `for`: Int?
```
Comment (vytautasgimbutas) [inline Sources/Entities/Allowance/PSAllowanceValid.swift]: let's name it `validFor`

File: Sources/Entities/Allowance/PSTransactionAllowance.swift [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import ObjectMapper
+
+public class PSTransactionAllowance: Mappable {
+    public var id: Int?
+    public var data: PSAllowance?
+    public var optional: Bool!
```
Comment (vytautasgimbutas) [inline Sources/Entities/Allowance/PSTransactionAllowance.swift]: interesting that optional is highlighted like this

File: Sources/Entities/Allowance/PSAllowance.swift [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import ObjectMapper
+
+public class PSAllowance: Mappable {
+    public var description: String?
+    public var currency: String!
+    public var maxPrice: Int?
+    public var valid: PSAllowanceValid?
+    public var limits: [PSAllowanceLimit]?
```
Comment (vytautasgimbutas) [inline Sources/Entities/Allowance/PSAllowance.swift]: public var limits = [PSAllowanceLimit]()

File: Sources/Entities/Allowance/PSTransactionAllowance.swift (hunk 2 above)
Comment (Pattio) [reply Sources/Entities/Allowance/PSTransactionAllowance.swift]: It is named like that in backend, however, I could rename it to `isOptional`, although I would still have to force unwrap it, as giving default value doesn't seem correct approach

**Pull request merged**